admin-data-explorer-play-server/src/main/scala/net/wiringbits/webapp/utils/admin/services/AdminService.scala

```diff
@@ -18,66 +17,75 @@ class AdminService @Inject() (
-      val referenceField = maybe.getOrElse("id")
-      TableReference(referencedTable = tableName, referenceField = referenceField)
-    }
-
     for {
       items <- Future.sequence {
         tableSettings.tables.map { settings =>
           val hiddenColumns = settings.hiddenColumns
           for {
-            tableFields <- databaseTablesRepository.getTableFields(settings.tableName)
-            tableReferences <- databaseTablesRepository.getTableReferences(settings.tableName)
-
-            visibleFields = tableFields.filterNot(field => hiddenColumns.contains(field.name))
-            fields = visibleFields.map { field =>
-              val fieldName = getFieldName(field.name, settings.primaryKeyField)
-              val editable = !settings.nonEditableColumns.contains(field.name)
-              val referencedTableName = getReferencedTable(tableReferences, field.name)
-              val reference = referencedTableName.map { getFieldReference }
-              TableField(name = fieldName, `type` = field.`type`, editable = editable, reference = reference)
+            tableColumns <- databaseTablesRepository.getTableColumns(settings.tableName)
+            foreignKeys <- databaseTablesRepository.getForeignKeys(settings.tableName)
+
+            visibleColumns = tableColumns.filterNot(column => hiddenColumns.contains(column.name))
+            columns = visibleColumns.map { column =>
+              val fieldName = getColumnName(column.name, settings.primaryKeyField)
+              val isEditable = !settings.nonEditableColumns.contains(column.name)
+              val reference = getColumnReference(foreignKeys, column.name)
+              AdminGetTables.Response.TableColumn(
+                name = fieldName,
+                `type` = column.`type`,
+                editable = isEditable,
+                reference = reference
+              )
             }
           } yield AdminGetTables.Response.DatabaseTable(
             name = settings.tableName,
-            fields = fields,
+            columns = columns,
             primaryKeyName = settings.primaryKeyField
           )
         }
       }
     } yield AdminGetTables.Response(items)
   }
 
-  def tableMetadata(
-      tableName: String,
-      queryParameters: QueryParameters
-  ): Future[(List[Map[String, String]], String)] = {
-    validateTableName(tableName)
-    val settings = tableSettings.unsafeFindByName(tableName)
-    val sortParam = queryParameters.sort.fromPrimaryKeyField(settings.primaryKeyField)
-    val queryParams = queryParameters.copy(sort = sortParam)
+  private def hideData(tableData: TableData, hiddenColumns: List[String]) = {
+    tableData.data.hideData(hiddenColumns)
+  }
+
+  def tableMetadata(tableName: String, queryParams: QueryParameters): Future[(List[Map[String, String]], String)] = {
+    val validations = {
+      for {
+        _ <- Future.successful(validateTableName(tableName))
```

Comment:
This needs to be `Future(validateTableName(...))` `because `Future.successful` expects constant values only and it won't handle exceptions